### ISSUE TITLE
xmpp text messages from jmp.chat cause node red to crash

### ISSUE THREAD

**Issue by mstyp** [node-red/node-red-nodes#897]
I am running node red [via yunohost](https://github.com/YunoHost-Apps/nodered_ynh). version 2.2.2~ynh1. 

When I recieve a regular xmpp message things work fine. When I recieve a text message via https://jmp.chat, node red service crashes. 

After seeing the logs the lead dev of jmp.chat said:
> looks like a bug in their stanza parser of some kind. I wonder if they are assuming JabberIDs have a resource. That's a guess

Here are the error logs from yunohost:

```
TIME node[30242]: 1 TIME - [red] Uncaught Exception:
TIME node[30242]: 1 TIME - [error] TypeError: Cannot read properties of undefined (reading 'length')
TIME node[30242]:     at Client.<anonymous> (/opt/yunohost/nodered/data/node_modules/node-red-node-xmpp/92-xmpp.js:447:36)
TIME node[30242]:     at Client.emit (node:events:538:35)
TIME node[30242]:     at Client._onElement (/opt/yunohost/nodered/data/node_modules/node-red-node-xmpp/node_modules/@xmpp/connection/index.js:105:10)
TIME node[30242]:     at Parser.emit (node:events:526:28)
TIME node[30242]:     at Parser.onEndElement (/opt/yunohost/nodered/data/node_modules/node-red-node-xmpp/node_modules/@xmpp/xml/lib/Parser.js:52:12)
TIME node[30242]:     at SaxLtx.emit (node:events:526:28)
TIME node[30242]:     at SaxLtx._handleTagOpening (/opt/yunohost/nodered/data/node_modules/node-red-node-xmpp/node_modules/ltx/lib/parsers/ltx.js:44:14)
TIME node[30242]:     at SaxLtx.write (/opt/yunohost/nodered/data/node_modules/node-red-node-xmpp/node_modules/ltx/lib/parsers/ltx.js:192:20)
TIME node[30242]:     at Parser.write (/opt/yunohost/nodered/data/node_modules/node-red-node-xmpp/node_modules/@xmpp/xml/lib/Parser.js:71:17)
TIME node[30242]:     at Client._onData (/opt/yunohost/nodered/data/node_modules/node-red-node-xmpp/node_modules/@xmpp/connection/index.js:49:17)
TIME systemd[1]: nodered.service: Main process exited, code=exited, status=1/FAILURE
TIME systemd[1]: nodered.service: Failed with result 'exit-code'.
```
In gajim, when I look at the raw xml that jmp.chat sends over when I get a text message, this is what they send:

```
<!-- Incoming DATE (MYJID) -->
<message xmlns="jabber:client" id="BIGLONGSTRING" type="chat" to="MYJID" from="+PHONENUMBER@cheogram.com">
  <origin-id id="BIGLONGSTRING" xmlns="urn:xmpp:sid:0" />
  <request xmlns="urn:xmpp:receipts" />
  <markable xmlns="urn:xmpp:chat-markers:0" />
  <body>text message</body>
  <stanza-id id="BIGLONGSTRING" by="MYJID" xmlns="urn:xmpp:sid:0" />
</message>```

```

**Comment by dceejay** [comment]
oops - yes - bad handling.

**Comment by rallep71** [comment]
Hello, a question:

Why can't I send images or audio files from Node Red with the Node Xmpp? It is possible in every XMPP messenger. 

With almost every other messenger node, whether Telegram...or others, it is possible to send pictures from Node-Red in a simple way.
Example: The doorbell rings, snapshot with the IP cam, send picture to XMPP client.


thx

**Comment by hardillb** [comment]
@rallep71 This has nothing to do with this now closed issue, if you have questions ask them on the Node-RED Forum, Node-RED Slack (both are linked to from the Node-RED homepage) or StackOverflow with a suitable tag. Issues are for defects in the code only.

**Comment by dceejay** [comment]
Short answer is that the xmpp.js library we use does not immediately support images in any meaningful way. So you need to either work around it by uploading the image to a server (could be Node-RED http end-point) then sending the link, or maybe base64 encoding it into the message somehow - or something else... Net is we would be happy to look at a Pull Request to add it but it isn't something that is supported by default XMPP.

**Comment by mstyp** [comment]
> oops - yes - bad handling. Fix in 0.5.10

Awesome! This is one of the big reasons I love FOSS. Every time a bug like this crops up the scars from the multitudes of unanswered / unresponsive bug reports I used to submit to MS and google flare up, and its responses like this (verify issue is real, implementing a patch) that help me overcome that hesitancy and open issues.

Who knows maybe all those old bug reports where counted in some behind the scenes way, but being able to see "ah yep, issue is real, fix coming" just wipes out all of the frustration I had associated with the bug.
